PyInstaller/lib/modulegraph/modulegraph.py

```diff
@@ -2339,6 +2339,24 @@ def _safe_import_hook(
         # Target module imported above.
         target_module = target_modules[0]
 
+        if (isinstance(target_module, MissingModule) and
+                source_module is not None and
+                target_attr_names is None and
+                level == ABSOLUTE_IMPORT_LEVEL and
+                type(source_module) is SourceModule and
+                target_module_partname ==
+                    '_' + source_module.identifier.rpartition('.')[2] and
+                sys.version_info[0] == 3):
```

Comment:
I suggest putting this into a function `is_swig_candidate()` (nested into `_safe_import_hook`, taking no parameters) an use it here and at line 2275 and here. This avoids problems by diverging code in the long run.